M/MPICH_CUDA/build_tarballs.jl

```diff
@@ -0,0 +1,158 @@
+# the Fortran bindings.  See for example
+# * https://stackoverflow.com/q/56759636/2442087
+# * https://github.com/pmodels/mpich/blob/d10400d7a8238dc3c8464184238202ecacfb53c7/doc/installguide/cfile
+export CROSS_F77_SIZEOF_INTEGER=4
+export CROSS_F77_SIZEOF_REAL=4
+export CROSS_F77_SIZEOF_DOUBLE_PRECISION=8
+export CROSS_F77_SIZEOF_LOGICAL=4
+export CROSS_F77_TRUE_VALUE=1
+export CROSS_F77_FALSE_VALUE=0
+
+if [[ ${nbits} == 32 ]]; then
+    export CROSS_F90_ADDRESS_KIND=4
+else
+    export CROSS_F90_ADDRESS_KIND=8
+fi
+export CROSS_F90_OFFSET_KIND=8
+export CROSS_F90_INTEGER_KIND=4
+export CROSS_F90_INTEGER_MODEL=9
+export CROSS_F90_REAL_MODEL=6,37
+export CROSS_F90_DOUBLE_MODEL=15,307
+export CROSS_F90_ALL_INTEGER_MODELS=2,1,4,2,9,4,18,8,
+export CROSS_F90_INTEGER_MODEL_MAP={2,1,1},{4,2,2},{9,4,4},{18,8,8},
+
+./configure "${configure_flags[@]}"
+
+# Remove empty `-l` flags from libtool
+# (Why are they there? They should not be.)
+# Run the command several times to handle multiple (overlapping) occurrences.
+sed -i 's/"-l /"/g;s/ -l / /g;s/-l"/"/g' libtool
+sed -i 's/"-l /"/g;s/ -l / /g;s/-l"/"/g' libtool
+sed -i 's/"-l /"/g;s/ -l / /g;s/-l"/"/g' libtool
+
+# Build the library
+make -j${nproc}
+
+# Install the library
+make install
+
+# Install the license
+install_license $WORKSPACE/srcdir/mpich*/COPYRIGHT
+"""
+
+
+platforms = CUDA.supported_platforms(; min_version = v"11.8", max_version=v"12.9.999")
+filter!(p -> arch(p) == "x86_64" || arch(p) == "aarch64", platforms)
+platforms = expand_gfortran_versions(platforms)
+
+# Add `mpi+mpich` platform tag
+for p in platforms
+    p["mpi"] = "MPICH"
+end
+
+products = [
+    # MPICH
+    LibraryProduct("libmpicxx", :libmpicxx),
+    LibraryProduct("libmpifort", :libmpifort),
+    LibraryProduct("libmpi", :libmpi),
+    ExecutableProduct("mpiexec", :mpiexec),
+]
+
+dependencies = [
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency("Hwloc_jll"; compat="2.12.0"),
+    RuntimeDependency(PackageSpec(name="MPIPreferences", uuid="3da0fdf6-3ccc-4f1b-acd9-58baa6c99267");
+                      compat="0.1", top_level=true),
+]
+
+
+for platform in platforms
+
+    should_build_platform(triplet(platform)) || continue
+
+    cuda_deps = CUDA.required_dependencies(platform, static_sdk=true)
+
+    cuda_ver = platform["cuda"]
+
+    platform_sources = BinaryBuilder.AbstractSource[sources...]
+
+    # Add x86_64 CUDA_SDK to cross compile for aarch64
+    if arch(platform) == "aarch64"
+        push!(platform_sources, CUDA.cuda_nvcc_redist_source(cuda_ver, "x86_64"))
+    end
+
+    build_tarballs(ARGS, name, version, platform_sources, script, [platform],
+                    products, [dependencies; cuda_deps];
+                    julia_compat = "1.6", clang_use_lld=false,
+                    preferred_gcc_version = v"5", augment_platform_block=CUDA.augment,
+                )
+end
```

Comment:
Missing newlines visually hurt me
```suggestion
end

```